### ISSUE TITLE
Detecting private fields in traits change broken with storeApis = false

### ISSUE THREAD

**Issue by lukaszwawrzyk** [sbt/zinc#600]
The PR https://github.com/sbt/zinc/pull/542 does not work with storeApis = false. Easiest reproduction is to add `incOptions.properties` with `incOptions.storeApis=false` to `trait-private-val` test.
This causes `java.lang.IllegalArgumentException: No companions was stored!` because at `sbt.internal.inc.IncrementalNameHashingCommon#findAPIChange` we are accessing the api to see if both classes are really traits. This api, obviously, is not stored and we fail.

I can submit a PR. The suggested solution would be to simply remove the guard and always return `Some(TraitPrivateMembersModified(className))` delaying the problem until some other case when `extraHash` differ is introduced. Let me know what you think @jvican

**Comment by jvican** [comment]
IMO skipping API storage is a red flag: `findAPIChange` is all about detecting changes at the API level and therefore the API information is paramount for a proper change detection.
  
If you still want to use that option, I'd propose that you only return the trait private members change if `incOptions.storeApis==true`, but honestly this is only delaying the problem and I won't personally have the bandwidth to make this work with future changes in the area. I don't see how the `storeApis=false` can be realistically be supported in the future. What is the motivation behind its use?

**Comment by lukaszwawrzyk** [comment]
The motivation behind this flag was the cache size. The cache for a part of the project including classes, analysis and apis is like 75% larger with `storeApis==true`. 
But what you are saying makes sense, we will enable the storeApis despite the size problem.
If the flag won't be supported perhaps it will be best to remove it.

**Comment by jvican** [comment]
> But what you are saying makes sense, we will enable the storeApis despite the size problem. If the flag won't be supported perhaps it will be best to remove it.
  
I think this is the right call. I have changes that I need to upstream to fix the problems with memory consumption and the use of API (in the future I'd like to remove it in such a way that all the current use cases work, including test frameworks -- which will need to be based on runtime reflection).
  
> The motivation behind this flag was the cache size.
  
What do you mean by cache size? You mean the size of the analysis in memory?

**Comment by lukaszwawrzyk** [comment]
I mean the zip file that is created with `FileAnalysisStore.binary(analysisFile).set(analysis)`. Though I just noticed that I compared the size of cache with stored apis and no compression against cache without stored apis and compressed. So the actual difference should be much smaller and it makes even more sense to use `storeApis==true`.

Well, the difference is 2.5%, not 75%. Not sure if the concern with space was valid back when it was decided not to store apis but now it is rather irrelevant.
